### DISCUSSION TITLE
Does it make sense to add Kamal to Gemfile?

### DISCUSSION THREAD

**Discussion by spryffee** [basecamp/kamal#739] [Q&A] [answered]
Considering I'm going to distribute Rails application so one could create own Kamal configuration and deploy, can there be any reason to have Kamal installed with development group of Gemfile as we was doing it with Capistrano?

**Comment by morgoth** [comment, marked as answer]
The reason might be to have control over kamal version.

**Comment by eric1234** [comment]
I've added it to my Gemfile when I wanted to use a fork of Kamal that had pending fixes.

**Comment by igor-alexandrov** [comment]
Since Kamal is an actual dependency of the application, but an infrastructure management tool, I don't like the idea of adding it to the application Gemfile. I usually have a separate Gemfile for it.
 
``` Gemfile
# ./kamal/Gemfile

# frozen_string_literal: true

source 'https://rubygems.org'

gem 'kamal', '~> 1.4.0'
```

``` bash
BUNDLE_GEMFILE=kamal/Gemfile bundle install
```

``` bash
BUNDLE_GEMFILE=kamal/Gemfile bundle binstub kamal --path ../bin
```

This will allow you to have `./bin/kamal`, manage Kamal's version with Rubygems and at the same time have your application's Gemfile clean.
